internal/engine/build_and_deployer_test.go

```diff
@@ -486,6 +493,38 @@ func TestCustomBuild(t *testing.T) {
 	}
 }
 
+func TestContainerBuildMultiStage(t *testing.T) {
+	f := newBDFixture(t, k8s.EnvDockerDesktop)
+	defer f.TearDown()
+
+	manifest := NewSanchoFastMultiStageManifest(f)
+	targets := buildTargets(manifest)
+	changed := f.WriteFile("a.txt", "a")
+	bs := resultToStateSet(alreadyBuiltSet, []string{changed}, f.deployInfo())
+
+	// There are two image targets, and only the second one is dirty
+	firstResult := store.BuildResult{Image: container.MustParseNamedTagged("sancho-base:tilt-prebuilt")}
+	bs[targets[0].ID()] = store.NewBuildState(firstResult, nil)
+
+	result, err := f.bd.BuildAndDeploy(f.ctx, f.st, targets, bs)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, 0, f.docker.BuildCount)
+	assert.Equal(t, 0, f.docker.PushCount)
+	assert.Equal(t, 1, f.docker.CopyCount)
+	assert.Equal(t, 1, len(f.docker.ExecCalls))
+	f.assertContainerRestarts(1)
+
+	// The BuildComplete action handler expects to get exactly one result
+	_, hasResult0 := result[manifest.ImageTargetAt(0).ID()]
+	assert.False(t, hasResult0)
+	_, hasResult1 := result[manifest.ImageTargetAt(1).ID()]
```

Comment:
🙌 

but why not `res, hasResult1 := ...` and use `res` below instead of calling `AsOneResult`?

Reply:
mostly because the upper.go call also uses AsOneResult()